apps/console/src/features/applications/pages/application-edit.tsx

```diff
@@ -115,6 +116,43 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
     const [ callBackIdpID, setcallBackIdpID ] = useState<string>();
     const [ callBackIdpName, setcallBackIdpName ] = useState<string>();
 
+    const {
+        data: application,
+        mutate: mutateApplicationGetRequest,
+        isLoading: isApplicationGetRequestLoading,
+        error: applicationGetRequestError
+    } = useGetApplication(applicationId, !!applicationId);
+
+    useEffect(() => {
+        setApplicationRequestLoading(isApplicationGetRequestLoading);
+    }, [ isApplicationGetRequestLoading ]);
```

Comment:
Can't we directly use the loading variable returned by the swr hook?

Reply:
There is a `setIsLoading` passed to the edit component and the nested form components to a very deep level and it is calling back to set the loading state of the parent. So, its not safe to refactor that at this time.

https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/applications/components/edit-application.tsx#L658